statistics/feedback.go

```diff
@@ -454,11 +455,61 @@ func getOverlapFraction(fb Feedback, bkt bucket) (float64, float64) {
 	return overlap, ratio
 }
 
+// mergeFullyContainedFeedback merges the max fraction of non-overlapped feedbacks that are fully contained in the bucket.
+func (b *BucketFeedback) mergeFullyContainedFeedback(sc *stmtctx.StatementContext, bkt bucket) (float64, float64, bool) {
+	var feedbacks []Feedback
+	// Get all the fully contained feedbacks.
+	for _, fb := range b.feedback {
+		res, err := outOfRange(sc, bkt.Lower, bkt.Upper, fb.Lower)
+		if res != 0 || err != nil {
+			return 0, 0, false
+		}
+		res, err = outOfRange(sc, bkt.Lower, bkt.Upper, fb.Upper)
+		if res != 0 || err != nil {
+			return 0, 0, false
+		}
+		feedbacks = append(feedbacks, fb)
+	}
+	// Sort feedbacks by end point,	then pick every feedback that is not overlapped
+	// with the previous chosen feedbacks.
+	var existsErr bool
+	sort.Slice(feedbacks, func(i, j int) bool {
+		res, err := feedbacks[i].Upper.CompareDatum(sc, feedbacks[j].Upper)
+		if err != nil {
+			existsErr = true
+		}
+		return res == -1
+	})
+	if existsErr {
+		return 0, 0, false
+	}
+	previousEnd := &types.Datum{}
+	var sumFraction, sumCount float64
+	for _, fb := range feedbacks {
+		res, err := previousEnd.CompareDatum(sc, fb.Lower)
+		if err != nil {
+			return 0, 0, false
+		}
+		if res <= 0 {
```

Comment:
Will it be possible that:
`bkt [1, 5]`
`fb1 [1, 3]` `fb2[4, 5]` `fb3[3, 5]`

If so, we should use `fb3` instead of `fb2` here considering *the max fraction of non-overlapped feedbacks*.

Reply:
Resolved by also sort by the start point.